File: python/simulation.py
import numpy as np
from scipy.integrate import odeint

from model.set_model import *

class Simulation(object):

    x = f_params()
    y0 = initial_values()

    tspan = range(1801)
    t = np.array(tspan)
    conditions = ['HRG10nM', 'HRG1nM', 'HRG01nM']

    RP        = np.empty((len(tspan), len(conditions)))
    ShP       = np.empty((len(tspan), len(conditions)))
    PI3K_act  = np.empty((len(tspan), len(conditions)))
    Raf_act   = np.empty((len(tspan), len(conditions)))
    MEKPP     = np.empty((len(tspan), len(conditions)))
    ERKPP     = np.empty((len(tspan), len(conditions)))
    Akt_PI_PP = np.empty((len(tspan), len(conditions)))

    for i, condition in enumerate(conditions):
        if condition == 'HRG10nM':
            y0[V.HRG] = 330. # 10nM
        elif condition == 'HRG1nM':
            y0[V.HRG] = 33.  # 1nM
        elif condition == 'HRG01nM':
            y0[V.HRG] = 3.3  # 0.1nM

        Y = odeint(diffeq, y0, tspan, args=tuple(x))

        RP[:,i] = 2 * (Y[:,V.RP] + Y[:,V.R_PI3K] + Y[:,V.R_PI3K_act] + 
                        Y[:,V.R_ShGS] + Y[:,V.R_ShP] + Y[:,V.R_Shc]) / y0[V.R] * 100.
        ShP[:,i] = Y[:,V.ShP] / y0[V.Shc] * 100.
        PI3K_act[:,i] = Y[:,V.PI3K_act] / y0[V.PI3K] * 100.
        Raf_act[:,i] = Y[:,V.Raf_act] / y0[V.Raf] * 100.
        MEKPP[:,i] = Y[:,V.MEKPP] / y0[V.MEK] * 100.
        ERKPP[:,i] = Y[:,V.ERKPP] / y0[V.ERK] * 100.
        Akt_PI_PP[:,i] = Y[:,V.Akt_PI_PP] / y0[V.Akt] * 100.

File: matlab/simulation.m
global f_var;

y0 = initialValues();
options = odeset('RelTol',1e-9,'AbsTol',1e-9);

t = 0:1:1800;

condition = 3;

sim_RP        = zeros(length(t),condition);
sim_ShP       = zeros(length(t),condition);
sim_PI3K_act  = zeros(length(t),condition);
sim_Raf_act   = zeros(length(t),condition);
sim_MEKPP     = zeros(length(t),condition);
sim_ERKPP     = zeros(length(t),condition);
sim_Akt_PI_PP = zeros(length(t),condition);

for i = 1:condition
    if i == 1
        y0(f_var.HRG) = 330.0; % 10nM
    elseif i == 2
        y0(f_var.HRG) = 33.0;  % 1nM
    elseif i == 3
        y0(f_var.HRG) = 3.3;   % 0.1nM
    end

    [T,Y] = ode15s(@diffeq,[0 1800],y0,options);

    sim_RP(:,i) = interp1(T,2*(Y(:,f_var.RP)+Y(:,f_var.R_PI3K)+Y(:,f_var.R_PI3K_act)+...
                  Y(:,f_var.R_ShGS)+Y(:,f_var.R_ShP)+Y(:,f_var.R_Shc))./y0(f_var.R).*100.0,t);
    sim_ShP(:,i) = interp1(T,Y(:,f_var.ShP)./y0(f_var.Shc).*100.0,t);
    sim_PI3K_act(:,i) = interp1(T,Y(:,f_var.PI3K_act)./y0(f_var.PI3K).*100.0,t);
    sim_Raf_act(:,i) = interp1(T,Y(:,f_var.Raf_act)./y0(f_var.Raf).*100.0,t);
    sim_MEKPP(:,i) = interp1(T,Y(:,f_var.MEKPP)./y0(f_var.MEK).*100.0,t);
    sim_ERKPP(:,i) = interp1(T,Y(:,f_var.ERKPP)./y0(f_var.ERK).*100.0,t);
    sim_Akt_PI_PP(:,i) = interp1(T,Y(:,f_var.Akt_PI_PP)./y0(f_var.Akt).*100.0,t);

end